packages/widgets/src/menubar.ts

```diff
@@ -382,6 +391,74 @@ export class MenuBar extends Widget {
     }
   }
 
+  /**
+   * A message handler invoked on an `'resize'` message.
+   */
+  protected _evtResize(event: Event): void {
+    let itemMenus = this.node.getElementsByClassName('lm-MenuBar-item');
+    let screenSize = this.node.offsetWidth;
+    let totalMenuSize = 0;
+    let index = -1;
+    let n = itemMenus.length;
+
+    if (this._menuItemSizes.length == 0) {
```

Comment:
I can move this block of code to another function but I would need to send almost all the variables defined in the function as a parameter, so I don't know what's best...

Reply:
It might just be me, but would it be better if the entire function were moved into a separate method? Like so:

```
_evtResize(event: Event): void {
  this._handleOverflow();
}
```

I'm not sure that's the best name for the method, but there are a few reasons why I think having such a method would be a good idea:

- I think it might be prudent to provide a way for clients to opt out of this behavior. I don't know all of the places that the Lumino menu bar class is used. I'm not 100% sure this behavior should be forced on downstream consumers always. Pulling the body of this function into its own method will make it easy to isolate that code and exit early if the client has passed some kind of "no overflow" option to the menu bar.
- Once we figure out how to hook into page load and where, we will be able to call this method (rather than `this._evtResize`) from there.
- Slightly simplified unit testing, since you don't have to trigger a resize event (also it makes it clear that you're not actually using the event object)
- Friendlier to future programmers if they want to add more stuff to the resize handler (that doesn't have to do with overflow)